src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy

```diff
@@ -227,112 +226,164 @@ class ActionDelegate {
+                            saveFieldBehavior(field, transitionObject, (behaviorClosureResult instanceof Set<FieldBehavior>) ? behaviorClosureResult : null)
+                        }
+                    } else if (transitionObject instanceof List<Transition>) {
+                        transitionObject.each { trans ->
+                            fields.each { field ->
+                                if (trans.dataSet.containsKey(field.stringId)) {
+                                    behaviorClosureResult = behavior(field, trans)
+                                    saveFieldBehavior(field, trans, (behaviorClosureResult instanceof Set<FieldBehavior>) ? behaviorClosureResult : null)
+                                }
                             }
                         }
+                    } else if (transitionObject instanceof Closure) {
+                        if (transitionObject == transitions) {
+                            useCase.petriNet.transitions.each {transitionEntry ->
+                                Transition trans = transitionEntry.value
+                                fields.each { field ->
+                                    if (trans.dataSet.containsKey(field.stringId)) {
+                                        behaviorClosureResult = behavior(field, trans)
+                                        saveFieldBehavior(field, trans, (behaviorClosureResult instanceof Set<FieldBehavior>) ? behaviorClosureResult : null)
+                                    }
+                                }
+                            }
+                        } else {
+                            throw new IllegalArgumentException("Invalid call of make method. Method call should contain specific transition (transitions) or keyword \'transitions\'.")
+                        }
+                    } else {
+                        throw new IllegalArgumentException("Invalid call of make method. Method call should contain specific transition (transitions) or keyword \'transitions\'.")
                     }
                 }
-            } else {
-                throw new IllegalArgumentException("Invalid call of make method. Method call should contain transition or keyword \'each\'.")
-            }
+            }]
         }]
     }
 
     protected SetDataEventOutcome createSetDataEventOutcome(){
         return new SetDataEventOutcome(this.useCase, this.task.orElse(null))
     }
 
-    def saveFieldBehavior(Field field, Transition trans) {
+    def saveFieldBehavior(Field field, Transition trans, Set<FieldBehavior> initialBehavior) {
         Map<String, Set<FieldBehavior>> fieldBehavior = useCase.dataSet.get(field.stringId).behavior
-        if (fieldBehavior.get(trans.getStringId()).find { it == FieldBehavior.INITIAL})
-            fieldBehavior.put(trans.stringId, useCase.petriNet.transitions.get(trans.stringId).dataSet.get(field.stringId).behavior)
+        if (initialBehavior != null)
```

Comment:
if `initialBehavior` is `null` then no behavior change happens? But the no-change event outcome is still emitted? Is this intended, or do I misunderstand the intentions?

Reply:
if initial behavior is null, behavior updates remains as implemented before (behavior set in "make" method will be set). If initial is not null, it replaces "initial" behavior from "make" method with original behavior